db/counting_index.go

```diff
@@ -103,71 +100,95 @@ func GetCountingIndex(kv KVStore, name []byte) (CountingIndex, error) {
 	return &countingIndex{
 		kvStore: kvRange,
 		bucket:  bucket,
-		size:    byteutil.BytesToUint64BigEndian(total),
 	}, nil
 }
 
 // Size returns the total number of keys so far
 func (c *countingIndex) Size() uint64 {
-	return atomic.LoadUint64(&c.size)
+	return c.size()
+}
+
+func (c *countingIndex) size() uint64 {
+	total, err := c.kvStore.Get(c.bucket, CountKey)
+	if errors.Cause(err) == ErrNotExist || total == nil {
+		return 0
+	}
+	return byteutil.BytesToUint64BigEndian(total)
 }
 
 // Add inserts a value into the index
-func (c *countingIndex) Add(value []byte, inBatch bool) error {
-	if inBatch {
-		return c.addBatch(value)
+func (c *countingIndex) AddOne(value []byte) error {
+	if err := c.TurnOn(); err != nil {
```

Comment:
the definition of `TurnOn` is the service is ready, which doesn't match the usage here.